pkg/snmp/traps/oid_resolver.go

```diff
@@ -85,11 +94,22 @@ func (or *MultiFilesOIDResolver) GetVariableMetadata(trapOID string, varOID stri
 	if !ok {
 		return VariableMetadata{}, fmt.Errorf("trap OID %s is not defined", trapOID)
 	}
-	varData, ok := trapData.variableSpecPtr[varOID]
-	if !ok {
-		return VariableMetadata{}, fmt.Errorf("variable OID %s is not defined", varOID)
+
+	varOIDSegments := strings.Split(varOID, ".")
+	for i := len(varOIDSegments); i > 0; i-- {
+		recreatedOID := strings.Join(varOIDSegments[0:i], ".")
+		varData, ok := trapData.variableSpecPtr[recreatedOID]
+		if !ok {
+			continue
+		}
+		if varData.isNode {
+			// Found a known Node while climibing up the tree, no chance of finding a match higher
+			return VariableMetadata{}, fmt.Errorf("variable OID %s is not defined", varOID)
+		}
+		return varData, nil
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏It might be more efficient to iterate with something like this below for long OIDs to avoid joining many segments many times (`strings.Join(varOIDSegments[0:i], ".")`):

```
for {
  lastDot := strings.LastIndex(".")
  if lastDot == -1 {
     break
  }
  newOID := varOID[0:]

  // break when string is empty
}
```